badger/cmd/write_bench.go

```diff
@@ -309,28 +312,35 @@ func reportStats(c *y.Closer, db *badger.DB) {
 		case <-c.HasBeenClosed():
 			return
 		case <-t.C:
-			txn := db.NewTransaction(false)
-			defer txn.Discard()
-
-			iopt := badger.DefaultIteratorOptions
-			iopt.AllVersions = true
-			iopt.InternalAccess = true
-
-			it := txn.NewIterator(iopt)
-			defer it.Close()
-			for it.Rewind(); it.Valid(); it.Next() {
-				i := it.Item()
-				if bytes.HasPrefix(i.Key(), []byte("!badger!")) {
-					internalKeyCount++
-				}
-				if bytes.HasPrefix(i.Key(), []byte("!badger!Move")) {
-					moveKeyCount++
-				}
-				if i.IsDeletedOrExpired() {
-					invalidKeyCount++
-				} else {
-					validKeyCount++
-				}
+			if showKeysCount {
+				func() {
+					txn := db.NewTransaction(false)
+					defer txn.Discard()
+
+					iopt := badger.DefaultIteratorOptions
+					iopt.AllVersions = true
+					iopt.InternalAccess = true
+					it := txn.NewIterator(iopt)
+					defer it.Close()
+
+					for it.Rewind(); it.Valid(); it.Next() {
+						i := it.Item()
+						if bytes.HasPrefix(i.Key(), []byte("!badger!")) {
+							internalKeyCount++
+						}
+						if bytes.HasPrefix(i.Key(), []byte("!badger!Move")) {
+							moveKeyCount++
+						}
+						if i.IsDeletedOrExpired() {
+							invalidKeyCount++
+						} else {
+							validKeyCount++
+						}
+					}
+				}()
+				fmt.Printf("Valid Keys Count: %d\nInvalid Keys Count: %d\nMove Keys Count:"+
```

Comment:
Remove the new lines. Make it look like.
```
Valid Keys: X Invalid Keys: Y Move Keys: Z Internal Key: A
```

Reply:
Done